devbox.go

```diff
@@ -313,17 +313,50 @@ func missingDevboxJSONError(dir string) error {
 			}
 		}
 	}
-	return usererr.New("No devbox.json found in %s, or any parent directories. Did you run `devbox init` yet?", dir)
+
+	parentDirCheckAddendum := ""
+	if didCheckParents {
+		parentDirCheckAddendum = ", or any parent directories"
+	}
+
+	return usererr.New("No devbox.json found in %s%s. Did you run `devbox init` yet?", dir, parentDirCheckAddendum)
 }
 
+// findConfigDir
 func findConfigDir(dir string) (string, error) {
 
 	// Sanitize the directory and use the absolute path as canonical form
-	cur, err := filepath.Abs(dir)
+	absDir, err := filepath.Abs(dir)
 	if err != nil {
 		return "", errors.WithStack(err)
 	}
 
+	// If the directory (`dir`) is specified, then we check just that directory.
+	// Otherwise, we search the parent directories.
+	if dir != "" {
+		fi, err := os.Stat(dir)
+		if err != nil {
+			return "", err
+		}
+		switch mode := fi.Mode(); {
+		case mode.IsDir():
+			if !plansdk.FileExists(filepath.Join(absDir, configFilename)) {
+				return "", missingDevboxJSONError(dir, false /*didCheckParents*/)
+			}
+			return absDir, nil
+		case mode.IsRegular(): // regular means 'file'
+			if !plansdk.FileExists(filepath.Clean(absDir)) {
+				return "", missingDevboxJSONError(dir, false /*didCheckParents*/)
+			}
+			// we return a directory from this function
+			return filepath.Dir(absDir), nil
+		default:
+			return "", errors.Errorf("unhandled mode %v", mode)
```

Comment:
What if we just assume that everything that isn't a directory is a file? If for some reason it is unsupported, then we'll get a read error later.

Reply:
sure :)